### ISSUE TITLE
[Feature Request] KEY_TO_GOAL_STRATEGY is useless

### ISSUE THREAD

**Issue by hbonnavaud** [DLR-RM/stable-baselines3#1627]
### 🚀 Feature

stable_baselines3.her.goal_selection_strategy.KEY_TO_GOAL_STRATEGY is useless.

The use of a dictionary to retrieve the Enum element is not clean, and requires to modify KEY_TO_GOAL_STRATEGY every a new goal selection strategy is added.

In the code, 
`KEY_TO_GOAL_STRATEGY[my_strategy]`
can be replaced by
`GoalSelectionStrategy[my_strategy.upper()]`
More, this replacement is more robust because a strategy written like "future" or "Future" or "FUTURE" will have the same result.

### Motivation

Make the implementation of HER more clean.

### Pitch

[Extract from the description]
In the code, 
`KEY_TO_GOAL_STRATEGY[my_strategy]`
can be replaced by
`GoalSelectionStrategy[my_strategy.upper()]`
More, this replacement is more robust because a strategy written like "future" or "Future" or "FUTURE" will have the same result.

### Alternatives

[Extract from the description]
In the code, 
`KEY_TO_GOAL_STRATEGY[my_strategy]`
can be replaced by
`GoalSelectionStrategy[my_strategy.upper()]`

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] If I'm requesting a new feature, I have proposed alternatives

**Comment by araffin** [comment]
Hello,
thanks for the proposal, I didn't know that this syntax worked.
```python
# In the code,
KEY_TO_GOAL_STRATEGY[my_strategy]
# can be replaced by
GoalSelectionStrategy[my_strategy.upper()]
```

I let @qgallouedec decide and we will probably welcome a PR then =)

**Comment by qgallouedec** [comment]
> I let @qgallouedec decide and we will probably welcome a PR then =)

I agree! However, I would just use `GoalSelectionStrategy[my_strategy]`, because I see no reason to force case-insensitivity in my opinion.

Feel free to submit a PR :)

**Comment by hbonnavaud** [comment]
I understand this, however, remove the .upper() implies that every strategy given as a string should match the exact name in the enum. Aka., everywhere occurence of "future" in SB3 code, should be replaced by "FUTURE", so it can create unexpected dumb errors that will force users to put their hands in the code ...

**Comment by hbonnavaud** [comment]
PS: in her_repplay_buffer.py line 76: `KEY_TO_GOAL_STRATEGY[goal_selection_strategy.lower()]`
You already had a ".lower()" for robustness. It's not that bad and I think it's a good practice when you manipulate strings as keys.

**Comment by qgallouedec** [comment]
> I understand this, however, remove the .upper() implies that every strategy given as a string should match the exact name in the enum. Aka., everywhere occurence of "future" in SB3 code, should be replaced by "FUTURE", so it can create unexpected dumb errors that will force users to put their hands in the code ...
> 
> 

It sounds legitimate after all
